packages/react-debug-tools/src/ReactDebugHooks.js

```diff
@@ -686,16 +713,25 @@ export function inspectHooksOfFiber(
   currentHook = (fiber.memoizedState: Hook);
   const contextMap = new Map();
   try {
-    setupContexts(contextMap, fiber);
-    if (fiber.tag === ForwardRef) {
+    if (fiber.tag === FunctionComponent) {
+      const legacyContext = setupContexts(
+        contextMap,
+        fiber,
+        enableLegacyContext,
+      );
+      return inspectHooks(type, props, currentDispatcher, legacyContext);
```

Comment:
Let's separate the code paths a little more explicitly?
```suggestion
    if (fiber.tag === FunctionComponent) {
      const hasLegacyContext = !!fiber.elementType.contextTypes;
      if (hasLegacyContext) {
        const legacyContext = getLegacyContext(fiber);
        return inspectHooks(type, props, currentDispatcher, legacyContext);
      } else {
        setupContext(contextMap, fiber);
        return inspectHooks(type, props, currentDispatcher);
      }
```